From b14f41aec34fe4c68471722acabf23e7a11ae163 Mon Sep 17 00:00:00 2001
From: Piotr Ziecik <kosmo@semihalf.com>
Date: Mon, 24 Mar 2014 16:34:33 +0100
Subject: [PATCH 1506/1825] dragonite: Setup Dragonite access to DDR.

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 295edfa751421b392ca8cb57f23e9b7cffa5646a

	This commit sets Dragonite memory map as follows:
	- R-Unit:	0x08000000	1MB
	- DFEV:		0x08100000	1MB
	- DDR CS[0]:	0x40000000	512MB

	Location of the Lantiq and R-Unit has been changed in
	order to leave space for large DDR window.

Change-Id: Ifdf5cebefd7d06aece81adf46f9ec6dc8fa30dfe
Signed-off-by: Piotr Ziecik <kosmo@semihalf.com>
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/6666
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_dragonite/dragonite.c        |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_dragonite/dragonite.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_dragonite/dragonite.c
index 5e40296..6ab2d77 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_dragonite/dragonite.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_dragonite/dragonite.c
@@ -36,9 +36,9 @@
 #define DRAGONITE_ITCM_SIZE	(80 * 1024)
 #define FIRMWARE_FILE		"dragonite/dragonite.bin"
 
-#define DRAGONITE_DRAM_BASE	0x30000000
-#define DRAGONITE_DFEV_BASE	0x40000000
-#define DRAGONITE_RUNIT_BASE	0x50000000
+#define DRAGONITE_RUNIT_BASE	0x08000000
+#define DRAGONITE_DFEV_BASE	0x08100000
+#define DRAGONITE_DRAM_BASE	0x40000000
 
 /* Dragonite Decoding Windows configuration */
 struct dragonite_win_cfg {
@@ -50,7 +50,7 @@ struct dragonite_win_cfg {
 };
 
 static struct dragonite_win_cfg dragonite_win_cfg[] = {
-	{ 0x00, 0x00, 0, SZ_256M, DRAGONITE_DRAM_BASE},	/* DDR */
+	{ 0x00, 0x0E, 0, SZ_512M, DRAGONITE_DRAM_BASE},	/* DDR */
 	{ 0x0B, 0x00, 0, SZ_1M,   DRAGONITE_DFEV_BASE},	/* Lantiq */
 	{ 0x01,	0x00, 1, SZ_1M,   DRAGONITE_RUNIT_BASE}, /* R-Unit */
 };
-- 
1.7.5.4

